src/clients/mocks/MockHubPoolClient.ts

```diff
@@ -0,0 +1,210 @@
+  proposer: "",
+  unclaimedPoolRebalanceLeafCount: 0,
+  challengePeriodEndTimestamp: 0,
+  bundleEvaluationBlockNumbers: [],
+  proposalBlockNumber: undefined,
+};
+
+export class MockHubPoolClient extends HubPoolClient {
+  public readonly minBlockRange = 10;
+  public rootBundleProposal = emptyRootBundle;
+
+  private events: Event[] = [];
+
+  private l1TokensMock: L1Token[] = []; // L1Tokens and their associated info.
+  private tokenInfoToReturn: L1Token = { address: "", decimals: 0, symbol: "" };
+  private l1TokensToDestinationTokensMock: { [l1Token: string]: { [destinationChainId: number]: string } } = {};
+  private returnedL1TokenForDeposit = "";
+  private eventManager: EventManager;
+
+  constructor(logger: winston.Logger, hubPool: Contract, configStoreClient: ConfigStoreClient, deploymentBlock = 0) {
+    super(logger, hubPool, configStoreClient, deploymentBlock);
+    this.eventManager = new EventManager(this.eventSignatures);
+  }
+
+  addEvent(event: Event): void {
+    this.events.push(event);
+  }
+
+  addL1Token(l1Token: L1Token) {
+    this.l1TokensMock.push(l1Token);
+  }
+
+  getL1Tokens() {
+    return this.l1TokensMock;
+  }
+
+  getTokenInfoForDeposit() {
+    return this.tokenInfoToReturn;
+  }
+
+  getTokenInfoForL1Token(l1Token: string): L1Token | undefined {
+    return this.l1TokensMock.find((token) => token.address === l1Token);
+  }
+
+  setTokenInfoToReturn(tokenInfo: L1Token) {
+    this.tokenInfoToReturn = tokenInfo;
+  }
+
+  setL1TokensToDestinationTokens(l1TokensToDestinationTokens: {
+    [l1Token: string]: { [destinationChainId: number]: string };
+  }) {
+    this.l1TokensToDestinationTokensMock = l1TokensToDestinationTokens;
+  }
+
+  getDestinationTokenForL1Token(l1Token: string, destinationChainId: number): string {
+    return (
+      this.l1TokensToDestinationTokensMock[l1Token]?.[destinationChainId] ??
+      super.getDestinationTokenForL1Token(l1Token, destinationChainId)
+    );
+  }
+
+  setReturnedL1TokenForDeposit(l1Token: string) {
+    this.returnedL1TokenForDeposit = l1Token;
+  }
+
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  getL1TokenForDeposit(_deposit: Deposit) {
+    return this.returnedL1TokenForDeposit ?? super.getL1TokenForDeposit(_deposit);
+  }
+
+  async _update(eventNames: string[]): Promise<HubPoolUpdate> {
+    // Generate new "on chain" responses.
+    const latestBlockNumber = (this.latestBlockNumber ?? 0) + random(this.minBlockRange, this.minBlockRange * 5, false);
+    const currentTime = Math.floor(Date.now() / 1000);
+
+    // Ensure an array for every requested event exists, in the requested order.
+    // All requested event types must be populated in the array (even if empty).
+    const _events: Event[][] = eventNames.map(() => []);
+    this.events.flat().forEach((event) => {
+      const idx = eventNames.indexOf(event.event as string);
+      if (idx !== -1) {
+        _events[idx].push(event);
+      }
+    });
+    this.events = [];
```

Comment:
This functionality (and everything following) was originally part of [this](https://github.com/across-protocol/relayer-v2/pull/646/files#diff-5f71e436c0671139f600446c187299f6940429e44531633234838dffbc7e4e20) PR in relayer-v2. That PR is held up on something unrelated to the test changes introduced to the HubPoolClient, so I'm taking the opportunity to carve those changes out and get them over here instead. It will reduce the scope of the residual work to be done in relayer-v2 (and it would have had to be migrated anyway).